api/envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -132,4 +132,12 @@ message LocalityLbEndpoints {
   //
   // Priorities should range from 0 (highest) to N (lowest) without skipping.
   uint32 priority = 5 [(validate.rules).uint32 = {lte: 128}];
+
+  // Optional: Per locality proximity value which indicates how close this
+  // locality is from the source locality. This value only provides ordering
+  // information (lower the value, closer it is to the source locality).
+  // This will be consumed by Load balancing schemes that need proximity order
+  // to determine where to route the requests.
+  // [#not-implemented-hide:]
+  google.protobuf.UInt32Value proximity_weight = 6;
```

Comment:
Not wanting to bikeshed, but wondering if we want this to be [0.0, 1.0] weight or an integer? Whatever is most useful is good with me.

Reply:
nit: can we call this just proximity or locality_proximity instead of proximity weight?

Reply:
This is semantically different than weight/pick-percent, integer values would better represent the absolute order [e.g. index of an array] we want to convey for these locality.

Changed to proximity